dataset_checker.py

```diff
@@ -0,0 +1,146 @@
+import torch
+import torch.nn as nn
+import torch.optim as optim
+import json
+import random
+import time
+from datetime import timedelta
+import glob
+import shutil
+
+
+class Model(nn.Module):
+    def __init__(self, input_size, hidden_dim1, hidden_dim2, target_size):
+        super(Model, self).__init__()
+        self.fc1 = nn.Linear(input_size, hidden_dim1)
+        self.fc2 = nn.Linear(hidden_dim1, hidden_dim2)
+        self.fc3 = nn.Linear(hidden_dim2, target_size)
+        self.softmax = nn.Softmax(dim=0)
+
+    def forward(self, landmarks):
+        x = self.fc1(landmarks)
+        x = self.fc2(x)
+        x = self.fc3(x)
+        res = self.softmax(x)
+        return res
+
+
+INPUT_SIZE = 42
+HIDDEN_DIM1 = 32
+HIDDEN_DIM2 = 32
+TARGET_SIZE = 8
+
+
+test_data_dir = "./dataset/test_data.json"
+data_dir = "./dataset/tmp/*"
+file_list = glob.glob(data_dir)
+file_list_json = [file for file in file_list if file.endswith(".json")]
+
+test_dataset = json.load(open(test_data_dir))
+
+
+def print_progress_bar(iteration, total, time_per_step, prefix="", suffix="", length=30, fill="="):
+    percent = ("{0:.1f}").format(100 * (iteration / float(total)))
+    filled_length = int(length * iteration // total)
+    bar = fill * filled_length + "-" * (length - filled_length)
+    time_str = str(timedelta(seconds=int(time_per_step * total)))
+    print(
+        f"\r{prefix} [{bar}] - {percent}% - {suffix} - {time_str}/epoch",
+        end="",
+        flush=True,
+    )
+
+
+for data_dir in file_list_json:
+    dataset = json.load(open(data_dir))
+    random.shuffle(dataset)
+    train_dataset = dataset[ : int(len(dataset) * 0.9) ]
+    val_dataset = dataset[ int(len(dataset) * 0.9) : ]
+
+    model = Model(INPUT_SIZE, HIDDEN_DIM1, HIDDEN_DIM2, TARGET_SIZE)
+    loss_function = nn.CrossEntropyLoss()
+    optimizer = optim.SGD(model.parameters(), lr=0.01)
+
+    total_epochs = 10
+    total_steps = len(train_dataset)
+
+    prev_loss = float("inf")
+    tolerance = 20
```

Comment:
total_epochs is 10. Could tolerance be 20?

Reply:
I think we have to figure out better way to early stop than tolerance